### DISCUSSION TITLE
How to use sleep / dormant

### DISCUSSION THREAD

**Discussion by drakejest** [earlephilhower/arduino-pico#344] [Q&A] [answered]
Hello I would like to sleep my rp2040, doing it the raspberry way of compiling all that is needed to have access to the sleep functions is to include 

`#include "pico/sleep.h"`

But of course the arduino IDE throws an error when i include that. I can already see the sleep library in the pico-extra, but i do not know how to include it in my sketch file. 

Thank you for the help

**Comment by drakejest** [comment, marked as answer]
I have not noticed immediately but for those also interested in these modes earl has instruction on how to do it here #345 

Quoting him

> In the short term I'm unfortunately occupied with other things, but to access the `pico-extra` sleep functions should be possible with some simple modifications to the core. (This is also for #344)
> 
> Basically, you will need to rebuild libpico.a to include the `pico_sleep` directory and add the proper include path to `platform_inc` to let the IDE fine the proper headers.
> 
>     1. Add the pico_sleep line to the Cmakefile.txt:   https://github.com/earlephilhower/arduino-pico/blob/master/tools/libpico/CMakeLists.txt
> 
>     2. Rebuild libpico.a by running make-libpico.sh  https://github.com/earlephilhower/arduino-pico/blob/master/tools/libpico/make-libpico.sh
> 
>     3. Add the pico-extras/pico_sleep path to pl;atform_inc.txt:  https://github.com/earlephilhower/arduino-pico/blob/master/lib/platform_inc.txt  (the format is a little off but should be obvious)
> 
> 
> Restarting the IDE should then give the ability to `#include <pico/sleep.h>` and call the functions exported there. Note that I have not looked deeply at this and it is in the `extras` repo so it may not be 100% working from the RPi folks,yet. I have a feeling you'll need to do things like save/restore all active clocks, set which clocks to disable in sleep mode, and make sure I/Os are set to appropriate drive strengths/inputs.

**Reply by yaswanthsk04** [reply]
Hi @drakejest :

Can you share me the pre-built core which includes sleep modes. 

I am pretty  new to the world of Cores and Compilation.

This would help me kick start me project as I am facing hard deadlines

Thanking you in Advance,
Yaswanthsk04
